### DISCUSSION TITLE
Feature Request: Preset builds with Linear Advance for v4.2.x no-mod boards?

### DISCUSSION THREAD

**Discussion by rollyceballos** [Jyers/Marlin#951] [Ideas]
Just a thought if we can have a preset build with Linear Advance for boards that may not need mods? I have been very satisfied with the preset builds e.g. UBL 10x10 high speed so thought maybe we can have a build for v4.2.x boards that may not need hardware mods when we have TMC2209?

**Comment by Jyers** [comment]
The issue with adding precompiled versions for linear advance is the fact that it would double my number of precompiled builds. I try to keep the versions to the most common configs and to a reasonable number as to not overwhelm newcomers and to make it easier to maintain. If you're interested in linear advance I would highly recommend looking into learning how to compile your own firmware as it adds a ton of customization and is definitely worth the effort. Or if you really hate visual studio you can always become a patreon and use the auto-compile service. Anyway, hope that all makes sense.

**Reply by rollyceballos** [reply]
No worries, I understand.  I do know how to build my own config and run the compile scripts.  But as my experience with ever changing config properties too over the Ender 3 V2 Marlin flavor it does take trial and error.  For example your personal config may be something I could start with but I know there are hardware mod config there that I should set back to previous values and takes me quite a while to find them.

Anyway may be a wiki for Linear advance can help? Prior to using your firmware I had LA custom config on the main Marlin branch but I’ve seen a lot more tweaks in your config that making my build had some surprises I’m still truing to get cross references to understand :)

**Reply by Jyers** [reply]
A wiki page on linear advance would be a great idea! There's actually some work going into a vastly improved wiki right now.

As far as what to change, if you don't use a mod you should be able to grab any config template and simply disable s-curve and enable linear advance. If you have any other questions about how to get it working feel free to let me know :)

**Reply by rollyceballos** [reply]
That’s great to know :D 

I’ll find those templates and see if I’d have a working one. Thanks!

again really appreciate your work @Jyers 👍

**Comment by PossibleCause** [comment]
Linear advance was one of my wishes to. I guess I will have to learn using visual code or become a patreon . Anyway thanks for all the work you nput into this. It is really Great!

**Comment by smooreace** [comment]
Its literally nothing be editing a "text" file. A 10-year-old with a basic
3d printing understanding can do it with ease. But... having an hourly.bin
file to fall back on if needed is a huge plus!

**Reply by rollyceballos** [reply]
LA is not something a 10-year old can fully comprehend as it can literally fry your drivers if done incorrectly…but agree it is a plus if there are preset builds :)
